### PR TITLE
feat(eventstore): previous aggregate root sequence

### PR DESCRIPTION
**Pull request by adlerhurst**

### CODE REVIEW

File: internal/eventstore/event.go [hunk 1]
```diff
@@ -36,7 +36,10 @@ type EventReader interface {
 
 	Sequence() uint64
 	CreationDate() time.Time
-	PreviousSequence() uint64
+	//PreviousAggregateSequence returns the previous sequence of the aggregate (e.g. for org.42508134)
+	PreviousAggregateSequence() uint64
+	//PreviousAggregateTypeSequence returns the previous sequence of the aggregate root (e.g. for org)
```
Comment (livio-a) [inline internal/eventstore/event.go:41]:
```suggestion
	//PreviousAggregateSequence returns the previous sequence of the aggregate root (e.g. for org.42508134)
	PreviousAggregateSequence() uint64
	//PreviousAggregateTypeSequence returns the previous sequence of the aggregate type (e.g. for org)
```

File: migrations/cockroach/V1.47__es_previous_agg_root_seq.sql [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+    set_previous sp 
+WHERE 
+    event_sequence = sp.agg_seq;
+
+
+-- --------------------------------------------------------
+-- org.domain
+-- --------------------------------------------------------
+WITH RECURSIVE set_previous (agg_type, agg_seq, agg_type_seq) AS (
+    (SELECT aggregate_type, MIN(event_sequence), NULL::INT8 FROM eventstore.events WHERE aggregate_type = 'org.domain' GROUP BY aggregate_type)
```
Comment (livio-a) [inline migrations/cockroach/V1.47__es_previous_agg_root_seq.sql:100]: there's no org.domain aggregate type

File: migrations/cockroach/V1.47__es_previous_agg_root_seq.sql [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+FROM 
+    set_previous sp 
+WHERE 
+    event_sequence = sp.agg_seq;
+
+
+-- --------------------------------------------------------
+-- org.name
+-- --------------------------------------------------------
+WITH RECURSIVE set_previous (agg_type, agg_seq, agg_type_seq) AS (
```
Comment (livio-a) [inline migrations/cockroach/V1.47__es_previous_agg_root_seq.sql:127]: org has already been done and org.name is no aggregate_type

File: migrations/cockroach/V1.47__es_previous_agg_root_seq.sql [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+    set_previous sp 
+WHERE 
+    event_sequence = sp.agg_seq;
+
+
+-- --------------------------------------------------------
+-- policy.password.complexity
+-- --------------------------------------------------------
+WITH RECURSIVE set_previous (agg_type, agg_seq, agg_type_seq) AS (
+    (SELECT aggregate_type, MIN(event_sequence), NULL::INT8 FROM eventstore.events WHERE aggregate_type = 'policy.password.complexity' GROUP BY aggregate_type)
```
Comment (livio-a) [inline migrations/cockroach/V1.47__es_previous_agg_root_seq.sql:156]: there's no policy.password.complexity aggregate type

File: migrations/cockroach/V1.47__es_previous_agg_root_seq.sql [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+    set_previous sp 
+WHERE 
+    event_sequence = sp.agg_seq;
+
+
+-- --------------------------------------------------------
+-- user.email
+-- --------------------------------------------------------
+WITH RECURSIVE set_previous (agg_type, agg_seq, agg_type_seq) AS (
+    (SELECT aggregate_type, MIN(event_sequence), NULL::INT8 FROM eventstore.events WHERE aggregate_type = 'user.email' GROUP BY aggregate_type)
```
Comment (livio-a) [inline migrations/cockroach/V1.47__es_previous_agg_root_seq.sql:240]: there's no user.email aggregate type

File: migrations/cockroach/V1.47__es_previous_agg_root_seq.sql [hunk 6]
```diff
@@ -0,0 +1,377 @@
...
+    set_previous sp 
+WHERE 
+    event_sequence = sp.agg_seq;
+
+
+-- --------------------------------------------------------
+-- user.human.externalidp
+-- --------------------------------------------------------
+WITH RECURSIVE set_previous (agg_type, agg_seq, agg_type_seq) AS (
+    (SELECT aggregate_type, MIN(event_sequence), NULL::INT8 FROM eventstore.events WHERE aggregate_type = 'user.human.externalidp' GROUP BY aggregate_type)
```
Comment (livio-a) [inline migrations/cockroach/V1.47__es_previous_agg_root_seq.sql:268]: there's no user.human.externalidp aggregate type

File: migrations/cockroach/V1.47__es_previous_agg_root_seq.sql [hunk 7]
```diff
@@ -0,0 +1,377 @@
...
+    set_previous sp 
+WHERE 
+    event_sequence = sp.agg_seq;
+
+
+-- --------------------------------------------------------
+-- user.username
+-- --------------------------------------------------------
+WITH RECURSIVE set_previous (agg_type, agg_seq, agg_type_seq) AS (
+    (SELECT aggregate_type, MIN(event_sequence), NULL::INT8 FROM eventstore.events WHERE aggregate_type = 'user.username' GROUP BY aggregate_type)
```
Comment (livio-a) [inline migrations/cockroach/V1.47__es_previous_agg_root_seq.sql:296]: there's no user.username aggregate type

File: migrations/cockroach/V1.47__es_previous_agg_root_seq.sql [hunk 8]
```diff
@@ -0,0 +1,377 @@
...
+    set_previous sp 
+WHERE 
+    event_sequence = sp.agg_seq;
+
+
+-- --------------------------------------------------------
+-- usergrant.unique
+-- --------------------------------------------------------
+WITH RECURSIVE set_previous (agg_type, agg_seq, agg_type_seq) AS (
+    (SELECT aggregate_type, MIN(event_sequence), NULL::INT8 FROM eventstore.events WHERE aggregate_type = 'usergrant.unique' GROUP BY aggregate_type)
```
Comment (livio-a) [inline migrations/cockroach/V1.47__es_previous_agg_root_seq.sql:352]: there's no usergrant.unique aggregate type

File: migrations/cockroach/V1.47__es_previous_agg_root_seq.sql (hunk 2 above)
Comment (adlerhurst) [reply migrations/cockroach/V1.47__es_previous_agg_root_seq.sql:100]: not anymore but there was one in the past

File: migrations/cockroach/V1.47__es_previous_agg_root_seq.sql (hunk 3 above)
Comment (adlerhurst) [reply migrations/cockroach/V1.47__es_previous_agg_root_seq.sql:127]: not anymore but there was one in the past

File: migrations/cockroach/V1.47__es_previous_agg_root_seq.sql (hunk 4 above)
Comment (adlerhurst) [reply migrations/cockroach/V1.47__es_previous_agg_root_seq.sql:156]: not anymore but there was one in the past

File: migrations/cockroach/V1.47__es_previous_agg_root_seq.sql (hunk 5 above)
Comment (adlerhurst) [reply migrations/cockroach/V1.47__es_previous_agg_root_seq.sql:240]: not anymore but there was one in the past

File: migrations/cockroach/V1.47__es_previous_agg_root_seq.sql (hunk 6 above)
Comment (adlerhurst) [reply migrations/cockroach/V1.47__es_previous_agg_root_seq.sql:268]: not anymore but there was one in the past

File: migrations/cockroach/V1.47__es_previous_agg_root_seq.sql (hunk 7 above)
Comment (adlerhurst) [reply migrations/cockroach/V1.47__es_previous_agg_root_seq.sql:296]: not anymore but there was one in the past

File: migrations/cockroach/V1.47__es_previous_agg_root_seq.sql (hunk 8 above)
Comment (adlerhurst) [reply migrations/cockroach/V1.47__es_previous_agg_root_seq.sql:352]: not anymore but there was one in the past

**Pull request merged**